internal/core/graphics/color.rs

```diff
@@ -51,17 +52,19 @@ pub struct Color {
     green: u8,
     blue: u8,
     alpha: u8,
+    // cache the conversion so additional method calls don't recalculate all the same values
+    hsva: Option<HsvaColor>,
```

Comment:
@tronical @ogoffart I'm unsure how best to handle this. My intention is to cache the HSV so that if multiple calls to the same color HSV are done it doesn't result in repetitive add/sub/div/mul of the same values. I don't know if that gets optimized out but it seems highly unlikely since the calls would result in distinct objects each time otherwise.